contracts/loan_pool/src/contract.rs

```diff
@@ -208,25 +74,31 @@ impl LoanPoolTrait for LoanPoolContract {
         // Extend instance storage rent
         extend_instance(e.clone());
 
-        // First transfer the pool shares that need to be redeemed
-        let share_token_client = token::Client::new(&e, &get_token_share(&e));
-        share_token_client.transfer(&user, &e.current_contract_address(), &amount);
+        // Get users receivables
+        let receivables_val: Val = positions::read_positions(&e, user.clone());
+        let receivables_map: Map<Symbol, i128> = Map::try_from_val(&e, &receivables_val).unwrap();
+        let receivables: i128 = receivables_map.get_unchecked(Symbol::new(&e, "receivables"));
 
-        let balance = get_balance_a(&e);
-        let balance_shares = get_balance_shares(&e);
-
-        let total_shares = get_total_shares(&e);
+        // Check that user is not trying to move more than receivables (TODO: also include collateral?)
+        assert!(
+            amount <= receivables,
+            "Amount can not be greater than receivables!"
+        );
```

Comment:
`assert` käyttö muualla kuin testissä. 🤔 

Saakohan näin tehdä / onko best practice?

Googlasin, on fine jos tässä tarkoitus panicoida. https://www.reddit.com/r/learnrust/comments/ew6op9/assert_outside_of_test_code/